File: library/src/main/java/com/stealthcopter/networktools/subnet/Device.java
package com.stealthcopter.networktools.subnet;

import java.net.InetAddress;

public class Device {
    public String ip;
    public String hostname;
    public String mac;
    public float time = 0;

    public Device(InetAddress ip) {
        this.ip = ip.getHostAddress();
        this.hostname = ip.getCanonicalHostName();
    }

    @Override
    public String toString() {
        return "Device{" +
                "ip='" + ip + '\'' +
                ", hostname='" + hostname + '\'' +
                ", mac='" + mac + '\'' +
                ", time=" + time +
                '}';
    }
}

File: library/src/main/java/com/stealthcopter/networktools/subnet/Device.kt
package com.stealthcopter.networktools.subnet

import java.net.InetAddress

class Device(ip: InetAddress) {
    @JvmField
    var ip: String
    var hostname: String
    @JvmField
    var mac: String? = null
    @JvmField
    var time = 0f

    init {
        this.ip = ip.hostAddress
        hostname = ip.canonicalHostName
    }

    override fun toString(): String {
        return "Device{" +
                "ip='" + ip + '\'' +
                ", hostname='" + hostname + '\'' +
                ", mac='" + mac + '\'' +
                ", time=" + time +
                '}'
    }
}